### PR TITLE
UF-7272 - Add municipality id to API

### PR DESCRIPTION
**Pull request by jklingst**
Adds `municipalityId` as a path parameter to all API endpoints. Also changes the HTTP status code for creating contracts from `204 NO CONTENT` to `201 CREATED` and modifies the endpoint for updating/patching a contract to return a `200 OK` with the updated contract as the body instead of `204 NO CONTENT`.

### CODE REVIEW

File: src/main/java/se/sundsvall/contract/api/ContractResource.java [hunk 1]
```diff
@@ -54,27 +58,34 @@
...
+	}
 
 	@Operation(
 		summary = "Create contract",
 		responses = {
 			@ApiResponse(
-				responseCode = "204",
-				description = "No content"
+				responseCode = "201",
+				description = "No content",
```
Comment (mha109) [inline src/main/java/se/sundsvall/contract/api/ContractResource.java:74]: Created right?

Comment (jklingst) [reply src/main/java/se/sundsvall/contract/api/ContractResource.java:74]: Yessir

Comment (jklingst) [reply src/main/java/se/sundsvall/contract/api/ContractResource.java:74]: Or "successful operation"

File: src/main/java/se/sundsvall/contract/api/ContractResource.java [hunk 2]
```diff
@@ -54,27 +58,34 @@
...
 			)
 		}
 	)
-	@PostMapping(produces = ALL_VALUE, consumes = APPLICATION_JSON_VALUE)
-	public ResponseEntity<Void> postLandLeaseContract(final UriComponentsBuilder uriComponentsBuilder,
-		@RequestBody @Valid final Contract contract) {
-
-		return ResponseEntity
-			.created(uriComponentsBuilder.build(service.createContract(contract)))
+	@PostMapping(consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_PROBLEM_JSON_VALUE)
```
Comment (mha109) [inline src/main/java/se/sundsvall/contract/api/ContractResource.java:84]: `produces = { APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE }` ?

**Review by mha109 — APPROVED** [review]

**Pull request merged**